consensus/test/stake_management_test.go

```diff
@@ -0,0 +1,326 @@
+package test
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"github.com/clearmatics/autonity/accounts/abi/bind"
+	"github.com/clearmatics/autonity/common"
+	"github.com/clearmatics/autonity/common/keygenerator"
+	"github.com/clearmatics/autonity/contracts/autonity"
+	"github.com/clearmatics/autonity/core"
+	"github.com/clearmatics/autonity/core/types"
+	"github.com/clearmatics/autonity/crypto"
+	"github.com/clearmatics/autonity/ethclient"
+	"math/big"
+	"strconv"
+	"sync"
+	"testing"
+)
+
+/*
+  In this file, it create 3 test cases which have similar work flow base on the local e2e test framework's main flow.
+
+  First it setup an autontiy network by according to the genesis hook function, then from the specific chain height, it
+  start to issue transactions via the transaction hook function specified for the target node, for example in the
+  mintStakeHook, redeemStakeHook, and sendStakeHook, it issues transaction to call autonity contract via operator account
+  to manage the stakes on the members.
+
+  Then the test case verify the output from its finalAssert hook function on the specified height of the blockchain, for
+  example, it checks the stake balance in different height to compare if the balance is expected.
+*/
+
+type testAutonity struct {
+	autonity       *Autonity
+	transactionOpt *bind.TransactOpts
+	callOpt        *bind.CallOpts
+	client         *ethclient.Client
+}
+
+func (a *testAutonity) Close() {
+	if a.client != nil {
+		a.client.Close()
+	}
+}
+
+func autonityInstance(operatorKey *ecdsa.PrivateKey, node *testNode) (*testAutonity, error) {
+
+	contract := new(testAutonity)
+
+	conn, err := ethclient.Dial("http://127.0.0.1:" + strconv.Itoa(node.rpcPort))
+	contract.client = conn
+	if err != nil {
+		return contract, err
+	}
```

Comment:
I am not sure why you want to set the client if Dial fails?

```suggestion
	conn, err := ethclient.Dial("http://127.0.0.1:" + strconv.Itoa(node.rpcPort))
	if err != nil {
		return contract, err
	}	
	contract.client = conn
```
```suggestion
	conn, err := ethclient.Dial("http://127.0.0.1:" + strconv.Itoa(node.rpcPort))
	contract.client = conn
	if err != nil {
		return contract, err
	}
```